src/realm/group_shared.cpp

```diff
@@ -1809,67 +1840,77 @@ void SharedGroup::do_end_read() noexcept
 }
 
 
-void SharedGroup::do_begin_write()
+bool SharedGroup::do_begin_write(bool must_wait)
```

Comment:
It seems to me that the majority of the code in this function is no longer on a shared path. This bool parameter then seems to change this function into two almost separate functions. Perhaps it would be better to keep the old, single-purpose function and move the new code path to the try version?